web_ui/CONTRIBUTING.md

```diff
@@ -0,0 +1,136 @@
+# Contributing to Intel Geti Web UI
+
+**Prerequisites**: We recommend developing Geti's UI locally using node v22.12.0 and npm v10.9.0. You will also need to have a Geti server available. Please see our [documentation]() for instructions to install Geti on your hardware.
+
+## Local Setup with npm 
+
+To get started with contributing to the UI first install all node dependencies locally
+
+```sh
+web_ui/ $ npm ci
+```
+## Setting up environment variables
+
+Duplicate .env.example file and rename it to .env
+Next make a `.env` file that includes a `REACT_APP_API_PROXY` variable, you copy over the `.env.example` file,
+```sh
+web_ui/ $ cp .env.example .env
+```
+
+Our proxy will redirect all `/api` requests  to `localhost:80/api` assuming your `REACT_APP_API_PROXY` is set to `localhost:80`. Update this variable with your server address.
+Assuming you have your Geti server running on `localhost:80` you are now all set to start changing the Geti UI.
+Start up the [Rsbuild](https://rsbuild.dev/) dev server with,
+
+```sh
+web_ui/ $ npm run start
+```
+
+The project should now load correctly.
+
+Keep reading to learn more about our linting and testing setup.
+
+## Testing and Quality Control
+
+We heavily rely on Typescript, ESLint, Jest and Playwright to make sure we don't introduce regressions as well as to help us improve the design of our code.
+
+### Typescript
+
```

Comment:
Thanks for your work on this, Mark!

For developers unfamiliar with the web-UI making changes in a different component, what's the recommended way to ensure their changes haven't introduced any regressions in Web-UI?

Given the existing Makefile structure and approach that cascades targets to component Makefiles, could we consider implementing [test targets](https://github.com/open-edge-platform/geti/blob/ef7c32d30f23ccd077dea8c2bb39a6c4462bd120/web_ui/Makefile#L16-L25) (perhaps wrapping npm commands) to easily trigger web-UI tests when running tests for the entire platform? This would be particularly helpful for external contributors and non-UI specialists.